packages/router/history.ts

```diff
@@ -36,17 +36,17 @@ export interface Path {
   /**
    * A URL pathname, beginning with a /.
    */
-  pathname: string;
+  pathname: `/${string}`;
 
   /**
    * A URL search string, beginning with a ?.
    */
-  search: string;
+  search: `?${string}`;
```

Comment:
just a thought.. this could also be: 
```suggestion
  search: `?${string}=${string}`;
```

thereby enforcing both `?` and `=` followed by strings